pkg/api/datasources.go

```diff
@@ -347,6 +347,18 @@ func validateJSONData(jsonData *simplejson.Json, cfg *setting.Cfg) error {
 			}
 		}
 	}
+
+	for key, value := range jsonData.Get("teamHeaders").MustMap() {
+		if strings.HasPrefix(key, datasources.CustomHeaderName) {
```

Comment:
This does not work, because in previous case it's assumed that data source json data has key started with `httpHeaderName`, but in our case we don't store headers this way. We need to go through all header records in the map and check if `header` field does not interfere with `AuthProxyHeaderName`.

Reply:
changed to only check the headers, you are right with the prefix here not being correct